### PR TITLE
Collections: fill, nCopies, addAll – and others how to quickly fill a collection?

### PR DESCRIPTION
**Pull request by Susfi**

### CODE REVIEW

File: src/main/java/com/epam/prejap/ess/helpers/collectionsfill/CollectionFillTimeTester.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+ * NCopies 11
+ * addAll(NCopies) 12
+ * Arrays.fill 16
+ * List.fill 19
+ * For loop 106
+ * stream 91
+ *
+ * clone(): The fastest way to fill a Collection with elements is to clone existing one, however we have no control
+ * over number of elements, nor what those elements are. We are also limited by what collections exist in the program.
+ * Clone() creates a shallow copy of Collection, so it does not creates copies of objects.
```
Comment (marcin-ogorzalek-EPAM) [inline src/main/java/com/epam/prejap/ess/helpers/collectionsfill/CollectionFillTimeTester.java:29]: I don't think that name of this methods is case insensitive.

File: src/main/java/com/epam/prejap/ess/helpers/collectionsfill/CollectionFillTimeTester.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        long startTime, endTime;
+        long timeArraysFill = 0, timeNCopies = 0, timeStream = 0, timeForLoop = 0, timeListFill = 0,
+                timeClone = 0, timeAddAll = 0, timeAddAllNCopies = 0;
+
+        for(int i = 0; i < NUMBER_OF_TRIES; i++) {
+
+            startTime = System.currentTimeMillis();
+            ArrayList<Object> arrayListFromArraysFill = new ArrayList<>(generateListUsingArraysFill());
+            endTime = System.currentTimeMillis();
+            timeArraysFill += endTime - startTime;
```
Comment (marcin-ogorzalek-EPAM) [inline src/main/java/com/epam/prejap/ess/helpers/collectionsfill/CollectionFillTimeTester.java:65]: It is acceptable in this case, but remember that creating those variable and assigning to them adds some overhead to the result.

File: src/main/java/com/epam/prejap/ess/helpers/collectionsfill/CollectionFillTimeTester.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+ * efficient
+ *
+ * Creating a list with for loop or stream are the slowest options, and they are memory inefficient as they create
+ * multiple copies of given object.
+ *
+ * Conclusion: the best way to create a list with n elements, or append existing list by n elements, is to use method
+ * nCopies() from Collections framework, unless desired effect is to have a list of copies of object.
+ * @author Paweł Susfał
+ */
+public class CollectionFillTimeTester {
```
Comment (marcin-ogorzalek-EPAM) [inline src/main/java/com/epam/prejap/ess/helpers/collectionsfill/CollectionFillTimeTester.java:51]: I think that this cals don't have to be `public`

**Comment by marcin-ogorzalek-EPAM** [comment]
Remember to bump up the project's version.

**Comment by marcin-ogorzalek-EPAM** [comment]
Pleas bump up version once more and we can marge it.

**Comment by marcin-ogorzalek-EPAM** [comment]
We have a merge conflict here. Please resolve it.

**Comment by Susfi** [comment]
@marcin-ogorzalek-EPAM conflicts resolved

**Comment by marcin-ogorzalek-EPAM** [comment]
Look at changed files here. You actually are pushing already introduced changes once more. Make sure that I'm going to merge only your files not all already present in the repo.

**Comment by TJB4epam** [comment]
Lecture code was added with small POM changes, please pull and update.

**Comment by Susfi** [comment]
@marcin-ogorzalek-EPAM conflict resolved

**Pull request merged**